infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/actions/RestoreRedundancyMergeSegments.java

```diff
@@ -170,22 +157,123 @@ Layout restoreWithBackOff(CorfuRuntime runtime, StateTransferManager transferMan
                 }
             } catch (TransferSegmentException e) {
                 throw new RetryExhaustedException("Transfer segment exception occurred.", e);
+            } catch (IllegalStateException e) {
+                throw new RetryExhaustedException("The current node is in the unresponsive list.", e);
             }
-
         }).setOptions(retrySettings).run();
+    }
 
+    /**
+     * Merge the state transfer segments and propose the new cluster layout.
+     *
+     * @param baseClient             Base client to the current node.
+     * @param layoutManagementView   Layout management view.
+     * @param layoutTransferSegments Layout and the transferred segments.
+     */
+    private void mergeSegments(BaseClient baseClient, LayoutManagementView layoutManagementView,
+                               LayoutTransferSegments layoutTransferSegments) {
+        Layout currentLayout = layoutTransferSegments.getLayout();
+        ImmutableList<TransferSegment> transferredSegments =
+                layoutTransferSegments.getTransferSegments();
+
+        // Seal the next epoch for the current server.
+        // This is done in order to prevent the unnecessary OutRankedExceptions during the
+        // layout reconfiguration that can happen if the min server set is sealed but the current node
+        // was not part of this set.
+        CFUtils.getUninterruptibly(baseClient.sealRemoteServer(currentLayout.getEpoch() + 1));
```

Comment:
I see, I think this problem is because the propose() method in LayoutView is not implemented in the best way. In this example, the propose should work because the majority can succeed, however in our propose() method, we uses CompleteableFuture.anyOf() and throws OutRankedException if we see any, which could be a false negative, when a majority can  succeed.

Reply:
Yeah, it makes sense but I don't want to touch Paxos code in this PR. I will create an issue for this:
https://github.com/CorfuDB/CorfuDB/issues/2574